jni-gen/systest/tests/field_setter_check_fails.rs

```diff
@@ -0,0 +1,85 @@
+use jni::objects::JObject;
+use jni::InitArgsBuilder;
+use jni::JNIVersion;
+use jni::JavaVM;
+use jni::JNIEnv;
+use jni::errors::Result as JNIResult;
+use systest::print_exception;
+use systest::wrappers::*;
+
+
+fn string_to_jobject<'a>(env: &JNIEnv<'a>, string: &str) -> JNIResult<JObject<'a>> {
+    Ok(JObject::from(env.new_string(string.to_owned())?))
+}
+
+#[test]
+#[should_panic(expected = "Java binding type failure. Expected object of class java/lang/Error, but got java/lang/Integer instead")]
+fn field_setter_should_fail_on_wrong_receiver() {
+    let jvm_args = InitArgsBuilder::new()
+        .version(JNIVersion::V8)
+        .option("-Xcheck:jni")
+        .option("-Xdebug")
+        .option("-XX:+CheckJNICalls")
+        //.option("-verbose:jni")
+        //.option("-XX:+TraceJNICalls")
+        .build()
+        .unwrap_or_else(|e| {
+            panic!("{} source: {:?}", e, std::error::Error::source(&e));
+        });
+
+    let jvm = JavaVM::new(jvm_args).unwrap_or_else(|e| {
+        panic!("{} source: {:?}", e, std::error::Error::source(&e));
+    });
+
+    let env = jvm
+        .attach_current_thread()
+        .expect("failed to attach jvm thread");
+
+    env.with_local_frame(16, || {
+        let error_wrapper = java::lang::Error::with(&env);
+        let integer_object = java::lang::Integer::with(&env).new(1337)?;
+        error_wrapper.set_detailMessage(integer_object, string_to_jobject(&env, "error message")?)?;
+        Ok(JObject::null())
+    }).unwrap_or_else(|e| {
+        print_exception(&env);
+        panic!("{} source: {:?}", e, std::error::Error::source(&e));
+    });
+}
+
+
+
+#[test]
+#[should_panic(expected = "Java binding type failure. Expected object of class java/lang/String, but got java/lang/Integer instead")]
+fn field_setter_should_fail_on_wrong_argument() {
+    let jvm_args = InitArgsBuilder::new()
+        .version(JNIVersion::V8)
+        .option("-Xcheck:jni")
+        .option("-Xdebug")
+        .option("-XX:+CheckJNICalls")
+        //.option("-verbose:jni")
+        //.option("-XX:+TraceJNICalls")
```

Comment:
Did you find all of some of these two options useful for debugging the tests? If not, could you remove those that you didn't need instead of commenting them?

Reply:
I just copied the code from https://github.com/viperproject/prusti-dev/blob/master/jni-gen/systest/tests/jvm_builtin_classes.rs#L26 without any changes. Removed in this version